lib/db/memory.js

```diff
@@ -269,7 +270,14 @@ MemoryStore.prototype = {
       if (client.canGrant === false) {
         if (! activeClients[clientIdHex]) {
           activeClients[clientIdHex] = client;
+          activeClients[clientIdHex].scope = [];
         }
+        clientTokenObj.clientScope.forEach(function (scope) {
```

Comment:
@divyabiyani This needs to handle this transformation:

with 3 token scopes of 1) `clients:write` 2) `profile profile:write` 3) `profile` you need to end up with: 

`clients:write profile profile:write`. 

Would this code here become:  `clients:write profile profile:write profile` ?